files/default/neutron-ha-tool.py

```diff
@@ -559,29 +574,88 @@ def migrate_router(qclient, router_id, agent_id, target_id):
     """
 
     LOG.info("Migrating router=%s from agent=%s to agent=%s",
-             router_id, agent_id, target_id)
+             router_id, agent['id'], target['id'])
 
     # N.B. The neutron API will return "success" even when there is a
     # subsequent failure during the add or remove process so we must check to
     # ensure the router has been added or removed
 
     # Remove the router from the original agent
-    qclient.remove_router_from_l3_agent(agent_id, router_id)
-    LOG.debug("Removed router from agent=%s" % agent_id)
+    qclient.remove_router_from_l3_agent(agent['id'], router_id)
+    LOG.debug("Removed router from agent=%s" % agent['id'])
 
     # ensure it is removed or log an error
-    if router_id in list_routers_on_l3_agent(qclient, agent_id):
+    if router_id in list_routers_on_l3_agent(qclient, agent['id']):
         raise RuntimeError("Failed to remove router_id=%s from agent_id=%s" %
-                           (router_id, agent_id))
+                           (router_id, agent['id']))
 
     # add the router id to a live agent
     router_body = {'router_id': router_id}
-    qclient.add_router_to_l3_agent(target_id, router_body)
+    qclient.add_router_to_l3_agent(target['id'], router_body)
 
     # ensure it is removed or log an error
-    if router_id not in list_routers_on_l3_agent(qclient, target_id):
+    if router_id not in list_routers_on_l3_agent(qclient, target['id']):
         raise RuntimeError("Failed to add router_id=%s from agent_id=%s" %
-                           (router_id, agent_id))
+                           (router_id, agent['id']))
+    if wait_for_router:
+        wait_router_migrated(qclient, router_id, target['host'])
+
+
+def wait_router_migrated(qclient, router_id, target_host, maxtries=60):
+    """
+    Returns nothing. Waits for all non-distributed ports and floating IPs
+    of a router for being in the ACTIVE again after a migration.
+
+    :param qclient: A neutron client
+    :param router_id: The id of the router to check
+    :param target_host: The host on which the ports should be ACTIVE
+                        (reflected by the binding:host_id attribute of
+                        a port)
+    :param maxtries: Maximum number of times the ports' status is polled,
+                     an expection is raised there are still ports with a
+                     non ACTIVE status after that.
+    """
+
+    LOG.info("Wait for the ports and floating IPs of router_id=%s "
+             "to be ACTIVE", router_id)
+    remaining_ports = ["dummy"]
+    remaining_fips = ["dummy"]
+    while maxtries:
```

Comment:
this block will always add a one second sleep, no matter how fast my ports become active, as initially both `remaining_fips` and `remaining_ports` are true. If I understand it correctly, the intention is to first check the ports, and then the floating ips, right? The other problem, is that the minimal value for `maxtries` should be 2, because one is taken by checking the ports, the other one is taken by checking the ips.

Let's use `continue` if we see that `remaining_ports` has emptied!

Also, could we add a separate variable instead of mutating `maxtries`, it is quite weird to read `maxtries -= 1`, something like this would read better:

```python
remaining_iterations = maxtries
while remaining_iterations:
    ...
    remaining_iterations -= 1
```